src/planscape/collaboration/tests/test_permissions.py

```diff
@@ -0,0 +1,235 @@
+from django.test import TransactionTestCase
+from django.contrib.auth.models import User
+from collaboration.models import Collaborator, Role
+from collaboration.utils import (
+    can_add_collaborators,
+    can_add_scenario,
+    can_archive_scenario,
+    can_change_collaborators,
+    can_delete_collaborators,
+    can_view_collaborators,
+    can_view_planning_area,
+    can_view_scenario,
+)
+from planning.models import PlanningArea, Scenario
+from django.contrib.contenttypes.models import ContentType
+
+
+class PermissionsTest(TransactionTestCase):
+    def setUp(self):
+        # create user
+        self.user = User.objects.create(username="testuser")
+        self.user.set_password("12345")
+        self.user.save()
+
+        # create second user (invitee?)
+        self.invitee = User.objects.create(username="invitee")
+        self.invitee.set_password("12345")
+        self.invitee.save()
+
+        # create planning area
+        self.planning_area = PlanningArea.objects.create(
+            user=self.user,
+            name="Test Planning Area",
+            region_name="sierra-nevada",
+            geometry=None,
+            notes="",
+        )
+        self.planning_area.save()
+
+        self.scenario = Scenario.objects.create(
+            user=self.user, planning_area=self.planning_area, name="a scenario"
+        )
+        self.scenario.save()
+
+    def create_collaborator_record(self, role: Role):
```

Comment:
Maybe we should turn this into a utility function outside of test, since we'll need something like this, anyway.